phpBB/includes/acp/acp_database.php

```diff
@@ -253,25 +323,51 @@ function main($id, $mode)
 
 								@set_time_limit(0);
 
-								$fp = @fopen($file_name, 'rb');
-
-								if ($fp !== false)
+								try
 								{
+									$fp = $storage->read_stream($file_name);
+
 									while (!feof($fp))
 									{
 										echo fread($fp, 8192);
 									}
 									fclose($fp);
 								}
+								catch (\phpbb\storage\exception\exception $e)
+								{
+
+								}
 
 								flush();
 								exit;
 							}
 
+							// Copy file to temp folder to decompress it
+							$temp_file_name = sys_get_temp_dir() . '/' . $file_name;
+
+							try
+							{
+								$stream = $storage->read_stream($file_name);
+								$fp = fopen($temp_file_name, 'w+b');
+
+								while (!feof($stream))
+								{
+									fwrite($fp, fread($stream, 8192));
+								}
+
+								fclose($fp);
+								fclose($stream);
+							}
+							catch (\phpbb\storage\exception\exception $e)
+							{
+								// Error while copying file from storage to temp
+								exit;
```

Comment:
An exit will not show any info to the user. I don't think that's a nice way of treating this exception.